autobahn/wamp/serializer.py

```diff
@@ -262,16 +288,24 @@ class JsonSerializer(Serializer):
     WAMP-over-Longpoll HTTP fallback.
     """
 
-    def __init__(self, batched=False):
+    def __init__(self, batched=False, muxed=False):
         """
         Ctor.
 
         :param batched: Flag to control whether to put this serialized into batched mode.
         :type batched: bool
         """
-        Serializer.__init__(self, JsonObjectSerializer(batched=batched))
+        Serializer.__init__(self, JsonObjectSerializer(batched=batched), muxed=muxed)
         if batched:
-            self.SERIALIZER_ID = u"json.batched"
+            if muxed:
+                self.SERIALIZER_ID = u"json.muxed.batched"
+            else:
+                self.SERIALIZER_ID = u"json.batched"
+        else:
+            if muxed:
+                self.SERIALIZER_ID = u"json.muxed"
+            else:
+                self.SERIALIZER_ID = u"json"
 
```

Comment:
All these "if" statements (for "batched" vs. "muxed") make me think there should be more classes instead, perhaps ... like a `JsonBatchedSerializer` and `JsonMuxedSerializer`, etc.